Modules/System/Cryptography Management/src/CryptographyManagementImpl.Codeunit.al

```diff
@@ -590,4 +590,24 @@ codeunit 1279 "Cryptography Management Impl."
         if IsNull(RijndaelProvider) then
             InitRijndaelProvider();
     end;
+
+    procedure HashRfc2898DeriveBytes(Password: Text; Salt: Text; NoOfBytes: Integer)
+    var
+        ByteArray: DotNet Array;
+        Convert: DotNet Convert;
+        Encoding: DotNet Encoding;
+        Rfc2898DeriveBytes: DotNet Rfc2898DeriveBytes;
+        SHA1Managed: DotNet SHA1Managed;
+    begin
+        if Salt = '' then
+            exit;
+
+        //Implement password-based key derivation functionality, PBKDF2, by using a pseudo-random number generator based on HMACSHA1.
+        Rfc2898DeriveBytes := Rfc2898DeriveBytes.Rfc2898DeriveBytes(Password, Encoding.ASCII.GetBytes(Salt));
+
+        //Return a Base64 encoded string of the SHA1 hash of the first X bytes (X = NoOfBytes) returned from the generator.
+        SHA1Managed := SHA1Managed.SHA1Managed;
+        ByteArray := SHA1Managed.ComputeHash(Rfc2898DeriveBytes.GetBytes(NoOfBytes));
```

Comment:
Remove SHA1Managed variable

if not TryGenerateHash(ByteArray, Rfc2898DeriveBytes.GetBytes(NoOfBytes), Format(HashAlgorithmType)) then
    Error(GetLastErrorText());